layer1/chain/src/chain.mligo

```diff
@@ -79,3 +128,43 @@ let find_latest_existing (chain : chain) : block option =
     let max_index = chain.max_index in
     find_existing max_index
 
+(* FINALIZATION *)
+(** auxialiary function: [get_last_elt (a::b::c::...::[z])] returns the last element in the list: (Some z) *)
+let rec get_last_elt (type a) (l : a list) : a option =
+    match l with
+    | [] -> (None : a option)
+    | t::[] -> Some t
+    | t::q -> get_last_elt q 
+
+(** [get_finalization_candidate_index (chain)] returns the index of a block, which is the first registered child of last finalized block *)
+let get_finalization_candidate_index (chain : chain) : index option =
+    match get_children (chain.latest_finalized, chain) with
+        | Some children -> get_last_elt children
```

Comment:
It's ok for now.
but one of assumptions for optimistic rollup is that the head of children is last added and it should be first finalization candidate.

Reply:
I added a FIXME to keep that in mind.